pydarn/io/borealis/borealis.py

```diff
@@ -0,0 +1,430 @@
+# Copyright 2019 SuperDARN Canada, University of Saskatchewan
+# Author: Marci Detwiller
+"""
+This file contains classes and functions for 
+converting of Borealis file types. 
+
+Classes
+-------
+BorealisConvert: Writes Borealis SuperDARN files types to
+SuperDARN DARN files with DMap record structure
+
+Exceptions
+----------
+BorealisFileTypeError
+BorealisFieldMissingError
+BorealisExtraFieldError
+BorealisDataFormatTypeError
+BorealisNumberOfRecordsError
+BorealisConversionTypesError
+BorealisConvert2IqdatError
+BorealisConvert2RawacfError
+BorealisRestructureError
+BorealisStructureError
+ConvertFileOverWriteError
+
+Functions
+---------
+read_borealis_file: uses BorealisArrayRead or BorealisSiteRead
+    and returns records or arrays (as requested)
+write_borealis_file: uses BorealisArrayWrite or BorealisSiteWrite
+    to write a file
+return_reader: attempts to read first as BorealisArrayRead and 
+    then as BorealisSiteRead and returns the reader class if one is 
+    successful
+borealis_site_to_array_file: uses BorealisSiteRead and BorealisArrayWrite
+    to convert a site file and write it to an array file.
+borealis_array_to_site_file: uses BorealisArrayRead and BorealisSiteWrite
+    to convert an array file and write it to a site file.
+borealis_write_to_dmap: uses read_borealis_file to read a file 
+    and BorealisConvert to then write a DARN file. 
+bfiq2darniqdat: reads a Borealis bfiq file and converts to DARN iqdat.
+rawacf2darnrawacf: reads a Borealis rawacf file and converts 
+    to DARN rawacf.
+
+See Also
+--------
+BorealisSiteRead
+BorealisSiteWrite
+BorealisArrayRead
+BorealisArrayWrite
+BorealisConvert
+
+For more information on Borealis data files and how they convert to DARN 
+filetypes, see: https://borealis.readthedocs.io/en/latest/ 
+"""
+
+import logging
+import os
+import sys
+import warnings
+
+from collections import OrderedDict
+from typing import Union, List
+
+from pydarn import borealis_exceptions, BorealisSiteRead, \
+                   BorealisSiteWrite, BorealisArrayRead, \
+                   BorealisArrayWrite, BorealisConvert, \
+                   BorealisRestructureUtilities
+
+pydarn_log = logging.getLogger('pydarn')
+
+
+def read_borealis_file(borealis_hdf5_file: str, borealis_filetype: str, 
+                       site: bool = False, records: bool = False) -> \
+                       Union[dict, OrderedDict]:
+    """
+    Read a Borealis file appropriately given site flag. Returns either
+    arrays or records according to records flag.
+
+    Parameters
+    ----------
+    borealis_hdf5_file
+        Borealis file to read. Either array or site style.
+    borealis_filetype
+        Borealis filetype. Possible types include antennas_iq, 
+        bfiq, rawacf, rawrf.
+    site: bool 
```

Comment:
`site` is usually a variable that asks for a name, not a flag.  This should be more descriptive (i.e., `site_flag` or `site_bool`)